src/dune_rpc_impl/server.ml

```diff
@@ -298,14 +305,45 @@ let handler (t : t Fdecl.t) : 'a Dune_rpc_server.Handler.t =
       let* running = Fiber.Pool.running t.pool in
       match running with
       | false -> Fiber.return ()
-      | true ->
-        let errors = Build_system.errors () in
-        Fiber.Pool.task t.pool ~f:(fun () ->
-            let events =
-              List.map errors ~f:(fun (e : Build_system.Error.t) ->
-                  Diagnostic.Event.Add (diagnostic_of_error e))
-            in
-            Session.notification session Server_notifications.diagnostic events)
+      | true -> (
+        match sub with
+        | Subscribe.Diagnostics ->
+          let errors = Build_system.errors () in
+          Fiber.Pool.task t.pool ~f:(fun () ->
+              let events =
+                List.map errors ~f:(fun (e : Build_system.Error.t) ->
+                    Diagnostic.Event.Add (diagnostic_of_error e))
+              in
+              Session.notification session Server_notifications.diagnostic
+                events)
+        | Build_progress ->
+          let current_progress : Build_system.Progress.t =
+            Build_system.get_current_progress ()
```

Comment:
What does this return if the build hasn't started yet?

Reply:
If I'm understanding the code correctly, this should give `0/0`, which I think is reasonable.

Reply:
Can't say I'm a fan of this result. It should not be displayed to the user, and the user has to reverse engineer the actual meaning behind it. I don't have a good idea on how to fix this yet, so I think we can accept this workaround for now.

Reply:
One alternative would be to merge `Build.Event` and `Progress` entirely, maybe something like

```
type t =
  | Waiting
  | In_progress of { done : int ; remaining : int }
  | Failed
  | Interrupted
  | Success
```

Reply:
That sounds reasonable to me.

Reply:
Do you think this is the right PR to do it in? That seems like a bit of a larger change.

Reply:
Another PR sounds more appropriate